hubblestack/extmods/grains/hostuuid.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+'''
+Generate a unique uuid for this host, storing it on disk so it persists across
+restarts
+'''
+import logging
+import os
+import uuid
+
+log = logging.getLogger(__name__)
+
+
+def host_uuid():
+    '''
+    Generate a unique uuid for this host, storing it on disk so it persists
+    across restarts
+    '''
+    cached_uuid_path = os.path.join(os.path.dirname(__opts__['configfile']), 'hubble_cached_uuid')
+    existing_uuid = __opts__.get('host_uuid', None)
+    try:
+        if os.path.isfile(cached_uuid_path):
+            with open(cached_uuid_path, 'r') as f:
+                cached_uuid = f.read()
+                # Check if it's changed out from under us -- problem!
+                if existing_uuid and cached_uuid != existing_uuid:
+                    log.error('host_uuid changed on disk unexpectedly!'
+                              '\nPrevious: {0}\nNew: {1}\nKeeping previous.'
+                              .format(existing_uuid, cached_uuid))
+                    # Write the previous UUID to the cache file
+                    try:
+                        with open(cached_uuid_path, 'w') as f:
+                            f.write(existing_uuid)
+                    except Exception:
+                        log.exception('Problem writing cached host uuid to file: {0}'
+                                      .format(cached_uuid_path))
+                    return {'host_uuid': existing_uuid}
+                return {'host_uuid': cached_uuid}
+        elif existing_uuid:
+            log.error('host_uuid was previously generated, but the cached '
+                      'file is no longer present: {0}'.format(cached_uuid_path))
+        else:
+            # TODO: once we figure out how to get a custom log level for
+            # non-error splunk logs, we should move this to that log level so
+            # it doesn't look like an error.
+            log.error('generating fresh uuid, no cache file found. '
```

Comment:
Is it possible to structure this so that when a new uuid is generated, it is included with the log message? 
The reason being that depending on how this is logged, on the siem side stitching together messages and figuring out a new uuid becomes harder. This is also one case when this field is not (yet) present when sent to a returner, so details would help.